### ISSUE TITLE
Players know more than they should in Prob end tournaments.

### ISSUE THREAD

**Issue by drvinceknight** [Axelrod-Python/Axelrod#610]
These two code snippets show that the two players defect on the last two rounds in both a normal tournament and a prob end tournament:

```
import axelrod as axl
import random

random.seed(0)
p1 = axl.strategy_transformers.FinalTransformer(['D', 'D'])(axl.Cooperator)()
p2 = axl.strategy_transformers.FinalTransformer(['D', 'D'])(axl.Cooperator)()
players = [p1, p2]
tournament = axl.Tournament(players, turns=5, repetitions=2)
results = tournament.play()
results.interactions
```

and 

```
random.seed(0)
p1 = axl.strategy_transformers.FinalTransformer(['D', 'D'])(axl.Cooperator)()
p2 = axl.strategy_transformers.FinalTransformer(['D', 'D'])(axl.Cooperator)()
players = [p1, p2]
tournament = axl.ProbEndTournament(players, prob_end=.1, repetitions=2)
results = tournament.play()
results.interactions
```

This is expected behaviour for the normal tournaments but not for the prob end.

I think, the fix is relatively straightforward once #609 is in. For the prob end tournament we just need to overwrite the `play_matches` method (although probably easier to pull out the creation of a match in there to another method) and pass a `match_attributes` dict that 'lies' to the players about the match length (setting it to `-1`).

**Comment by drvinceknight** [comment]
Something like that second code snippet should be a test for the prob end tournament.

**Comment by meatballs** [comment]
That might be better fixed in the match generator

**Comment by drvinceknight** [comment]
By adding the `match_attributes` dict to the output of `build_single_match_params`? 

That could be an even simpler fix...

**Comment by meatballs** [comment]
Yep, that's exactly what I had in mind

**Comment by drvinceknight** [comment]
I think that's way better than my suggestion. Potentially a one line fix right?

**Comment by drvinceknight** [comment]
Fixed by #609
